dts/arm/silabs/efr32bg22.dtsi

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2021 Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv8-m.dtsi>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+#include "gpio_gecko.h"
+
+/ {
+	chosen {
+		zephyr,flash-controller = &msc;
+		zephyr,entropy = &trng;
+	};
+
+	power-states {
+		standby: standby {
+			compatible = "zephyr,power-state";
+			power-state-name = "standby";
+			min-residency-us = < 50000 >;
+		};
+	};
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		cpu0: cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m33";
+			reg = <0>;
+			cpu-power-states = < &standby >;
+		};
+	};
+
+	sram0: memory@20000000 {
+		compatible = "mmio-sram";
+	};
+
+	soc {
+		msc: flash-controller@40030000 {
+			compatible = "silabs,gecko-flash-controller";
+			label = "FLASH_CTRL";
+			reg = <0x40030000 0xC69>;
+			interrupts = <49 0>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			flash0: flash@0 {
+				compatible = "soc-nv-flash";
+				label = "FLASH_0";
+				write-block-size = <4>;
+				erase-block-size = <8192>;
+			};
+		};
+
+		usart0: usart@4005C000 { /* USART0 */
+			compatible = "silabs,gecko-usart";
+			reg = <0x4005C000 0x400>;
+			interrupts = <13 0>, <14 0>;
+			interrupt-names = "rx", "tx";
+			peripheral-id = <0>;
+			status = "disabled";
+			label = "USART_0";
+		};
+
+		usart1: usart@40060000 { /* USART1 */
+			compatible = "silabs,gecko-usart";
+			reg = <0x40060000 0x400>;
+			interrupts = <15 0>, <16 0>;
+			interrupt-names = "rx", "tx";
+			peripheral-id = <1>;
+			status = "disabled";
+			label = "USART_1";
+		};
+
+		trng: trng@4C021000 {
+			compatible = "silabs,gecko-trng";
+			reg = < 0x4C021000 0x1000 >;
+			status = "disabled";
+			interrupts = < 0x1 0x0 >;
+			label = "TRNG0";
+		};
+
+
+		leuart0: leuart@4004a000 { /* LEUART0 */
+			compatible = "silabs,gecko-leuart";
+			reg = <0x4004a000 0x400>;
+			interrupts = <22 0>;
+			peripheral-id = <0>;
+			status = "disabled";
+			label = "LEUART_0";
+		};
+
+		i2c0: i2c@4a010000 {
+			compatible = "silabs,gecko-i2c";
+			clock-frequency = <I2C_BITRATE_STANDARD>;
+			reg = <0x4a010000 0x3044>;
+			interrupts = <27 0>;
+			label = "I2C_0";
+			status = "disabled";
+		};
+
+		i2c1: i2c@40068000 {
+			compatible = "silabs,gecko-i2c";
+			clock-frequency = <I2C_BITRATE_STANDARD>;
+			reg = <0x40068000 0x3044>;
+			interrupts = <28 0>;
+			label = "I2C_1";
+			status = "disabled";
+		};
+
+		stimer0: stimer@48000000 {
+			compatible = "silabs,gecko-stimer";
+			reg = <0x48000000 0x3054>;
+			interrupts = <12 0>;
+			clock-frequency = <32768>;
+			prescaler = <1>;
+			status = "disabled";
+			label = "STIMER_0";
+		};
+
+		gpio: gpio@0x4003C000 {
```

Comment:
Please remove the leading 0x from the node names. I get a some warnings, when i build, e.g. hello_world:

```
west build --pristine -b efr32bg_sltb010a samples/hello_world/
...
efr32bg_sltb010a.dts.pre.tmp:134.27-142.6: Warning (unit_address_format): /soc/gpio@0x4003C000/gpio@0x4003C000: unit name should not have leading "0x"
  also defined at efr32bg_sltb010a.dts.pre.tmp:286.8-288.3
efr32bg_sltb010a.dts.pre.tmp:143.27-151.6: Warning (unit_address_format): /soc/gpio@0x4003C000/gpio@0x4003C030: unit name should not have leading "0x"
  also defined at efr32bg_sltb010a.dts.pre.tmp:289.8-291.3
efr32bg_sltb010a.dts.pre.tmp:152.27-160.6: Warning (unit_address_format): /soc/gpio@0x4003C000/gpio@0x4003C060: unit name should not have leading "0x"
  also defined at efr32bg_sltb010a.dts.pre.tmp:292.8-294.3
efr32bg_sltb010a.dts.pre.tmp:161.27-169.6: Warning (unit_address_format): /soc/gpio@0x4003C000/gpio@0x4003C090: unit name should not have leading "0x"
  also defined at efr32bg_sltb010a.dts.pre.tmp:295.8-297.3
efr32bg_sltb010a.dts.pre.tmp:170.27-178.6: Warning (unit_address_format): /soc/gpio@0x4003C000/gpio@0x4003C0C0: unit name should not have leading "0x"
  also defined at efr32bg_sltb010a.dts.pre.tmp:298.8-300.3
efr32bg_sltb010a.dts.pre.tmp:67.26-75.5: Warning (simple_bus_reg): /soc/usart@4005C000: simple-bus unit address format error, expected "4005c000"
  also defined at efr32bg_sltb010a.dts.pre.tmp:239.9-244.3
efr32bg_sltb010a.dts.pre.tmp:85.23-91.5: Warning (simple_bus_reg): /soc/trng@4C021000: simple-bus unit address format error, expected "4c021000"
efr32bg_sltb010a.dts.pre.tmp:125.25-179.5: Warning (simple_bus_reg): /soc/gpio@0x4003C000: simple-bus unit address format error, expected "4003c000"
  also defined at efr32bg_sltb010a.dts.pre.tmp:282.7-285.3
efr32bg_sltb010a.dts.pre.tmp:180.24-187.5: Warning (simple_bus_reg): /soc/wdog@4A018000: simple-bus unit address format error, expected "4a018000"
  also defined at efr32bg_sltb010a.dts.pre.tmp:251.8-253.3
  also defined at efr32bg_sltb010a.dts.pre.tmp:301.8-303.3
...

```

Reply:
Fixed